src/app/core/xmpp/xmpp.service.ts

```diff
@@ -66,24 +67,40 @@ export class XmppService {
    * (meaning the user who is connected to the xmpp server)
    */
   public isJidCurrentUser(bareJid: string): Promise<boolean> {
-    return this._client.then((client) => client.jid.bare === bareJid);
+    return this.getClient().then((client) => client.jid.bare === bareJid);
   }
 
   /**
    * Returns a promise for a client with a working connection.
    * Automatically tries to connect if no connection is established.
    */
   public getClient(): Promise<any> {
-    return this._client.then(client => new Promise(resolve => {
+    return new Promise((resolve) => {
       if (this._state === ConnectionState.Up) {
-        resolve(client);
+        resolve(this._client);
+      } else if (this._state === ConnectionState.Down) {
+        // Register specific callbacks to see if reconnecting fails.
+        const errCallback = () => {
+          this.notificationService.alert(
+            'Connection lost',
+            'Multiple attempts to connect to the XMPP server have failed. ' +
+            'To retry, reload the page.',
+            false);
+        };
+        const succCallBack = () => {
+          // unsubscribe callbacks
+          this._client.off('session:started', succCallBack);
+          this._client.off('disconnected', errCallback);
+          resolve(this._client);
+        };
+        this._client.on('session:started', succCallBack);
```

Comment:
Yes, the guarantee is the XmppFeatureGuardService. It loads the config service that then sets the config on the xmpp service...As I quickly mentioned on the phone, I'm not super happy about this but otherwise this would cascade through the entire application which is non-sense as the config must be present before anything else..